padertorch/contrib/examples/source_localization/distance_estimator/data.py

```diff
@@ -0,0 +1,459 @@
+                Length (in samples) of the microphone signals to be generated
+            random_speech_samples:
+                If True, random source signal samples are used for the creaton
+                of the microphone signals
+        """
+        self.speech_set = None
+        if audio_key is None:
+            msg = "To generate the audio data, a path to a source signal" \
+                  " database is needed"
+            assert source_sig_path is not None, msg
+            speech_db_part = kwargs.get('speech_db_part', 'train_clean_100')
+            assert speech_db_part in ["dev_clean", "test_clean",
+                                      "train_clean_100", "train_clean_360"]
+            self.speech_set = \
+                JsonDatabase(source_sig_path).get_dataset(speech_db_part)
+        self.random_speech_samples = random_speech_samples
+        if source_sig_path is None:
+            msg = "The key under which the audio recording can be found in " \
+                  "the JSON of the database is required."
+            assert audio_key is not None, msg
+        self.signal_length = signal_length
+        self.mic_pair = kwargs.get('mic_pair')
+        self.audio_key = audio_key
+
+    def __call__(self, example):
+        if self.speech_set is not None:
+            rirs = self.prepare_rirs(example)
+            speech = self.prepare_speech(example, rirs.shape[-1])
+            mic_signals = reverb_signal(speech, rirs)
+        else:
+            # use of an own database is assumed,
+            # where the creation of microphone signals is not needed
+            mic_signals = self.prepare_recording(example)
+        return mic_signals
+
+    def prepare_rirs(self, example):
+        rirs = load_audio(example['rir'], dtype=np.float32)
+        assert rirs.shape[0] >= 2
+        if rirs.shape[0] > 2:
+            if 'mic_pair' in example.keys():
+                mic_pair = example['mic_pair']
+                rirs = rirs[np.asarray(mic_pair)]
+            else:
+                rirs = rirs[np.asarray(self.mic_pair)]
+        return rirs
+
+    def prepare_speech(self, example, len_rirs):
+        min_len = self.signal_length + len_rirs - 1
+        start_offset = 0
+        end = -1
+        if self.random_speech_samples:
+            speech_key = str_to_random_generator(
+                example["example_id"]).choice(len(self.speech_set))
+            counter = 0
+        while end <= min_len + start_offset:
+            if self.random_speech_samples:
+                speech_sample = self.speech_set[speech_key + counter]
+                counter += 1
+            else:
+                speech_sample = self.speech_set.random_choice()
+            end = speech_sample['offset']
+            start_offset = speech_sample['onset']
+        audio_path = speech_sample['audio_path']['observation']
+        audio = load_audio(audio_path, dtype=np.float32)
+        slice_start = np.random.randint(start_offset, end - min_len)
```

Comment:
if self.random_speech_samples is False, the random number generator should also be fixed using the exmaple_id (similar to the sampling of speech_key).